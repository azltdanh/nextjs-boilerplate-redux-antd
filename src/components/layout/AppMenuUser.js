import { signIn, useSession } from 'next-auth/client';
import { useRouter } from 'next/router';
import { Avatar, Button, Menu, Dropdown } from 'antd';
import {
  UserOutlined,
  ProfileOutlined,
  SettingOutlined,
  LogoutOutlined,
} from '@ant-design/icons';

const SubMenu = ({ user }) => {
  const router = useRouter();
  return (
    <Menu>
      <Menu.Item disabled>{user?.email}</Menu.Item>
      <Menu.Item
        icon={<ProfileOutlined />}
        key="/profile"
        onClick={() => router.push('/profile')}
      >
        Profile
      </Menu.Item>
      <Menu.Item
        icon={<SettingOutlined />}
        key="/setting"
        onClick={() => router.push('/setting')}
      >
        Setting
      </Menu.Item>
      <Menu.Item
        icon={<LogoutOutlined />}
        key="/log-out"
        onClick={() => router.push('/log-out')}
      >
        Log out
      </Menu.Item>
    </Menu>
  );
};

const AppMenuUser = () => {
  const [session] = useSession();
  const router = useRouter();

  return (
    <>
      {!session && (
        <>
          <Button type="default" size="small" onClick={signIn}>
            SIGN IN
          </Button>
          <Button
            type="default"
            size="small"
            onClick={() => router.push('/sign-up')}
          >
            SIGN UP
          </Button>
        </>
      )}
      {session && (
        <>
          <Dropdown
            overlay={<SubMenu user={session.user} />}
            placement="bottomRight"
          >
            <Avatar icon={<UserOutlined />} />
          </Dropdown>
        </>
      )}
    </>
  );
};

export default AppMenuUser;
