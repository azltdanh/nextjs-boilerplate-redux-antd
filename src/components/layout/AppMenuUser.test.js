import AppMenuUser from './AppMenuUser';

const mockHistoryPush = jest.fn();

jest.mock('next/router', () => ({
  ...jest.requireActual('next/router'),
  useRouter: () => ({
    pathname: '/',
    push: mockHistoryPush,
  }),
}));

describe(AppMenuUser.name, () => {
  it('renders', () => {
    const wrapper = shallow(<AppMenuUser />);
    expect(wrapper).toMatchSnapshot();

    // const items = wrapper.find('MenuItem');
    // expect(items.length).toEqual(3);

    // items.at(0).simulate('click');
    // expect(mockHistoryPush).toHaveBeenCalledWith('/log-in');

    // items.at(1).simulate('click');
    // expect(mockHistoryPush).toHaveBeenCalledWith('/log-out');

    // items.at(2).simulate('click');
    // expect(mockHistoryPush).toHaveBeenCalledWith('/sign-up');
  });
});
