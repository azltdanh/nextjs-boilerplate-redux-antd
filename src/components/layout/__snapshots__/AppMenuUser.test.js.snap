// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppMenuUser renders 1`] = `
<Fragment>
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[Function]}
    size="small"
    type="default"
  >
    SIGN IN
  </Button>
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[Function]}
    size="small"
    type="default"
  >
    SIGN UP
  </Button>
</Fragment>
`;
